### PR TITLE
fix: Add workaround for unsupported 'cache' field in Cloudflare Workers

### PR DESCRIPTION
**Pull request by anku255**
## Summary of change

Add workaround for unsupported 'cache' field in Cloudflare Workers

If a fetch request fails due to the 'cache' field not being implemented in Cloudflare Workers, retry the request without the 'cache' field. More info: https://github.com/cloudflare/workerd/issues/698

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `lib/ts/version.ts`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [x] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [x] Had run `npm run build-pretty`
-   [x] Had installed and ran the pre-commit hook
-   [ ] If new thirdparty provider is added,
    -   [ ] update switch statement in `recipe/thirdparty/providers/configUtils.ts` file, `createProvider` function.
    -   [ ] add an icon on the user management dashboard.
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `add-ts-no-check.js` file to include that
-   [ ] If added a new recipe / api interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [ ] If added a new recipe, then make sure to expose it inside the recipe folder present in the root of this repo. We also need to expose its types.

### CODE REVIEW

File: examples/cloudflare-workers/with-emailpassword/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+![SuperTokens banner](https://raw.githubusercontent.com/supertokens/supertokens-logo/master/images/Artboard%20%E2%80%93%2027%402x.png)
+
+# SuperTokens EmailPassword Demo app
```
Comment (rishabhpoddar) [inline examples/cloudflare-workers/with-emailpassword/README.md:3]: incorrect readme content

File: examples/cloudflare-workers/with-emailpassword/test/basic.test.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+ */
+
+/*
+ * Imports
+ */
+
+const assert = require("assert");
+const puppeteer = require("puppeteer");
+const { getTestEmail, setInputValues, submitForm, toggleSignInSignUp } = require("../../../../test/exampleTestHelpers");
+
```
Comment (rishabhpoddar) [inline examples/cloudflare-workers/with-emailpassword/test/basic.test.js:23]: does this test actually run in a cloudflare worker setup>

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Comment by ngoquoc** [comment]
Not sure if it's because of recent changes in Cloudflare runtime or not, but the error returned since couple of days ago:
```
Error: The 'cache' field on 'RequestInitializerDict' is not implemented.
```

It seems that the error is not thrown where Supertoken catches as I debugged and it could never reach `catch` block. I tried the following and it works:
```
const fetchFunction = typeof fetch !== "undefined" ? fetch : crossFetch;
    try {
        delete init.cache;
        return fetchFunction(input, init);
    } catch (e) {
        // it could never reach here
    }
```

@rishabhpoddar @anku255  Would it make sense to have a setting flag to enable/disable adding `cache` to `init` object behavior, so users have control over which behavior to use?

**Comment by anku255** [comment]
@ngoquoc I am not able to reproduce this. Can you share your `supertokens-node` and `wrangler` version? 

We also have a [cf-example app](https://github.com/supertokens/supertokens-node/tree/master/examples/cloudflare-workers/with-emailpassword). You can try that and see if you are able to reproduce this issue there.

**Comment by ngoquoc** [comment]
@anku255 
I'm using `"supertokens-node": "^18.0.0"` and `"wrangler": "^3.60.3"`.

I also deployed the example app (respecting the versions in package-lock) and could reproduce the same error. Please see the full error log:
[cf-init-cache-error-log.json](https://github.com/user-attachments/files/15880752/cf-init-cache-error-log.json)

**Comment by anku255** [comment]
Thank you @ngoquoc 

It seems that error is producible after deployment but not locally using the Wrangler CLI.  I will be taking a took soon. Will keep you updated on the same thread.

**Comment by anku255** [comment]
@ngoquoc We have released a fix in v18.0.1. You can try that and let us know.

**Comment by ngoquoc** [comment]
@anku255 thank you very much. Tried 18.0.1 and it works perfectly.

**Pull request merged**